crates/turbopack-dev-server/src/source/combined.rs

```diff
@@ -22,13 +23,26 @@ impl ContentSource for CombinedContentSource {
         path: &str,
         data: Value<ContentSourceData>,
     ) -> Result<ContentSourceResultVc> {
-        for source in self.sources.iter() {
-            let result = source.get(path, data.clone());
-            if !matches!(&*result.await?, ContentSourceResult::NotFound) {
-                return Ok(result);
-            }
-        }
-        Ok(ContentSourceResult::NotFound.cell())
+        let matches = self
+            .sources
+            .iter()
+            .map(|source| {
+                let result = source.get(path, data.clone());
+                async move { Ok((result, result.await?.specificity.await?)) }
+            })
+            .try_join()
```

Comment:
I'm worried that this will cause a all paths to be as slow as our slowest source (currently the node render source). Can we keep the old iteration, and early exit when an exact specificity is hit?

Reply:
yeah, good point, we can early exit